src/dune_engine/process.ml

```diff
@@ -378,7 +378,9 @@ module Exit_status = struct
       | None -> Path.of_string (Sys.getcwd ())
       | Some dir -> dir
     in
-    User_error.raise paragraphs ~annot:(With_directory_annot.make dir)
+    raise
```

Comment:
Why was this changed?

Reply:
On main, it was originally `raise`. I had initially changed it to `User_error.raise`, but that was leading to the CI failures where `Error:` got prepended to things, so I reverted the change.

Reply:
I do think that we should standardize the way that we raise user errors (either via `User_error.raise` or raising `User_error.E`), but that seems outside the scope of this PR.